phoenix-core/src/main/java/org/apache/phoenix/schema/json/PhoenixJson.java

```diff
@@ -0,0 +1,355 @@
+            return this.jsonAsString;
+        } else if (this.rootNode.isContainerNode()) {
+            return this.jsonAsString;
+        }
+
+        return null;
+
+    }
+
+    @Override
+    public String toString() {
+        return this.jsonAsString;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + this.jsonAsString.hashCode();
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        throw new EqualityNotSupportedException(PJson.INSTANCE);
+    }
+
+    /**
+     * @return length of the string represented by the current {@link PhoenixJson}.
+     */
+    public int estimateByteSize() {
+        String jsonStr = toString();
+        return jsonStr == null ? 1 : jsonStr.length();
+    }
+
+    public byte[] toBytes() {
+        return Bytes.toBytes(this.jsonAsString);
+    }
+
+    @Override
+    public int compareTo(PhoenixJson o) {
+        throw new EqualityNotSupportedException(PJson.INSTANCE);
+    }
+
+    private PhoenixJson getPhoenixJsonInternal(String[] paths) {
+        JsonNode node = this.rootNode;
+        for (String path : paths) {
+            JsonNode nodeTemp = null;
+            if (node.isArray()) {
+                int index = Integer.parseInt(path);
+                nodeTemp = node.path(index);
+            } else {
+                nodeTemp = node.path(path);
+            }
+            if (nodeTemp == null || nodeTemp.isMissingNode()) {
+                return null;
+            }
+            node = nodeTemp;
+        }
+        return new PhoenixJson(node, node.toString());
+    }
+
+    public int getJsonArrayLength() {
+        int count = 0;
+        Iterator<JsonNode> elements = this.rootNode.getElements();
+        while(elements.hasNext()){
+            elements.next();
+            count++;
+        }
+        return count;
+    }
+
+    public Object[] getJsonArrayElements() {
+        List<String> elementlist = new ArrayList();
+        Iterator<JsonNode> elements = this.rootNode.getElements();
+        while(elements.hasNext()){
+            JsonNode e = elements.next();
+            elementlist.add(e.toString());
+        }
+        return elementlist.toArray();
+    }
+
+    public Object[] getJsonObjectKeys() {
+        List<String> elementlist = new ArrayList();
+        Iterator<String> fieldnames = this.rootNode.getFieldNames();
+        while(fieldnames.hasNext()){
+            elementlist.add(fieldnames.next());
+        }
+        return elementlist.toArray();
+    }
+
+    public Object[] getJsonFields() {
+        List<String> elementlist = new ArrayList();
+        Iterator<Map.Entry<String, JsonNode>> fields = this.rootNode.getFields();
+
+        while(fields.hasNext()){
+            String fieldsstr = "";
+            Map.Entry<String, JsonNode> entry = fields.next();
+            fieldsstr += entry.getKey() +","+entry.getValue().toString();
```

Comment:
If you define String variable here, you do not need to use "+" operator, which is a costly affair avoid using it as much as possible.